### PR TITLE
fix: remove legacy source tracking commands

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
Removes the legacy versions of the Pull, Push and View Changes commands. Pretty much reverts most of the changes introduced with #3839 

### What issues does this PR fix or reference?
@W-12947655@

### Functionality Before
The following commands were visible/usable:
- SFDX: Pull Source from Default Scratch Org (Legacy)
- SFDX: Pull Source from Default Scratch Org and Override Conflicts (Legacy)
- SFDX: Push Source to Default Scratch Org (Legacy)
- SFDX: Push Source to Default Scratch Org and Override Conflicts (Legacy)
- SFDX: View All Changes (Local and in Default Scratch Org) (Legacy)
Pasted Graphic 1

### Functionality After
The following commands are not visible/usable:
- SFDX: Pull Source from Default Scratch Org (Legacy)
- SFDX: Pull Source from Default Scratch Org and Override Conflicts (Legacy)
- SFDX: Push Source to Default Scratch Org (Legacy)
- SFDX: Push Source to Default Scratch Org and Override Conflicts (Legacy)
- SFDX: View All Changes (Local and in Default Scratch Org) (Legacy)
Pasted Graphic

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/commands/forceSourcePull.ts [hunk 1]
```diff
@@ -68,9 +58,8 @@ const workspaceChecker = new SfdxWorkspaceChecker();
 const parameterGatherer = new EmptyParametersGatherer();
 
 export async function forceSourcePull(this: FlagParameter<string>) {
-  const { flag, commandVersion } = this || {};
-  const command =
-    commandVersion === CommandVersion.Legacy ? pullCommandLegacy : pullCommand;
+  const { flag } = this || {};
+  const command = pullCommand;
```
Comment (randi274) [inline packages/salesforcedx-vscode-core/src/commands/forceSourcePull.ts:62]: I think we can collapse this further and strike line 62, just passing `pullCommand` into `ForceSourcePullExecutor`

File: packages/salesforcedx-vscode-core/src/commands/forceSourcePush.ts [hunk 2]
```diff
@@ -73,9 +63,8 @@ const workspaceChecker = new SfdxWorkspaceChecker();
 const parameterGatherer = new EmptyParametersGatherer();
 
 export async function forceSourcePush(this: FlagParameter<string>) {
-  const { flag, commandVersion } = this || {};
-  const command =
-    commandVersion === CommandVersion.Legacy ? pushCommandLegacy : pushCommand;
+  const { flag } = this || {};
+  const command = pushCommand;
```
Comment (randi274) [inline packages/salesforcedx-vscode-core/src/commands/forceSourcePush.ts:67]: Same here - drop 67 and pass in pushCommand ⚡

File: packages/salesforcedx-vscode-core/src/commands/forceSourceStatus.ts [hunk 3]
```diff
@@ -80,11 +72,8 @@ const parameterGatherer = new EmptyParametersGatherer();
 export async function forceSourceStatus(
   this: FlagParameter<SourceStatusFlags>
 ) {
-  const { flag, commandVersion } = this || {};
-  const command =
-    commandVersion === CommandVersion.Legacy
-      ? statusCommandLegacy
-      : statusCommand;
```
Comment (randi274) [inline packages/salesforcedx-vscode-core/src/commands/forceSourceStatus.ts:87]: And ditto for statusCommand too :)

**Review by randi274 — CHANGES_REQUESTED** [review]
Looking real good @CristiCanizales! Builds are passing, and a quick spot check of the push/pull commands is working too.

I noticed only one extra place for cleanup. I saw CommandVersion is exported from sfdxCommandlet (and was originally added for the purpose of adding the legacy commands!) but is no longer being used. Can you remove that enum and subsequent references over there?

**Comment by CristiCanizales** [comment]
Sure! Nice catch. Thanks!

**Pull request merged**